template-only-bin/download-and-install-app.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# -----------------------------------------------------------------------------
+# This script downloads and installs infrastructure for an application.
+# Run this script in your project's root directory.
+#
+# Positional parameters:
+#   APP_NAME (required) - the name for the application, use kebab-case
+# -----------------------------------------------------------------------------
+set -euo pipefail
+
+APP_NAME=$1
+CURRENT_VERSION=$(cat .template-version)
+
+# Enforce kebab-case
+APP_NAME_KEBAB=$(echo "$APP_NAME" | tr "_" "-")
```

Comment:
i agree the convention should be dasherized, but i don't know how i feel about silently coercing it since if we normalize here we'll have to normalize in a lot of other places, and i don't think it should be the template's responsibility to do that. if we want to prevent underscores i'd rather just error out here. or even simpler i am ok with just leaving it as a convention but not actually preventing someone from using underscores.